relay/session.go

```diff
@@ -169,6 +170,57 @@ func (s *Server) CompleteSession(sessionID, localPartyID string) error {
 	return nil
 }
 
+func (s *Server) CheckCompletedParties(sessionID string, partiesJoined []string) (bool, error) {
+	sessionURL := s.vultisigRelay + "/complete/" + sessionID
+	start := time.Now()
+	timeout := time.Minute
+
+	for {
+		req, err := http.NewRequest(http.MethodGet, sessionURL, nil)
+		if err != nil {
+			return false, fmt.Errorf("fail to check completed parties: %w", err)
+		}
+
+		req.Header.Set("Content-Type", "application/json")
+		resp, err := s.client.Do(req)
+		if err != nil {
+			return false, fmt.Errorf("fail to check completed parties: %w", err)
+		}
+		if resp.StatusCode != http.StatusOK {
+			return false, fmt.Errorf("fail to check completed parties: %s", resp.Status)
+		}
+		defer resp.Body.Close()
+
+		result, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return false, fmt.Errorf("fail to fetch request: %w", err)
+		}
+
+		if len(result) > 0 {
+			var peers []string
+			err := json.Unmarshal(result, &peers)
+			if err != nil {
+				logging.Logger.WithFields(logrus.Fields{
+					"error": err,
+				}).Error("Failed to decode response to JSON")
+				continue
+			}
+
+			if common.IsSubset(peers, partiesJoined) {
```

Comment:
This check is wrong, instead you should check `peers` contains `partiesJoined` 
which means all parties finished keygen